wrapper/src/main/java/software/amazon/jdbc/hostlistprovider/AuroraHostListProvider.java

```diff
@@ -333,11 +337,27 @@ protected List<HostSpec> queryForTopology(final Connection conn) throws SQLExcep
       this.topologyAwareDialect = (TopologyAwareDatabaseCluster) this.hostListProviderService.getDialect();
     }
 
+    int networkTimeout = -1;
+    try {
+      networkTimeout = conn.getNetworkTimeout();
+      // The topology query is not monitored by the EFM plugin, so it needs a socket timeout
+      if (networkTimeout == 0) {
```

Comment:
Should we override the user's socket timeout here even if it is set to a non-zero value?

Reply:
What is the motivation for that?

Reply:
We should apply default 5000ms socket timeout just in case the current one is set to indefinite wait (it doesn't matter if this indefinite was a user explicit setting or just lack of setting)

Reply:
If they've set their socket timeout to accommodate their longest running queries it could be awhile before a problem is detected with the topology query. The topology query itself should be pretty quick to return. However, I lean towards keeping the code as is, I'd prefer to abide by the settings they gave us as much as possible. Just thought I'd bring it up to see if you agree

Reply:
@sergiyvamz won't the value returned by getNetworkTimeout be 0 regardless of if the user explicitly set it to zero or just didn't set it at all?

Reply:
I assume it is. But it makes sense to verify.

Reply:
I just checked and the default value is 0 with all drivers